packages/stylist-macros/src/tokentree/css_ident.rs

```diff
@@ -0,0 +1,122 @@
+use convert_case::{Case, Casing};
+use proc_macro2::{Punct, Spacing, TokenStream};
+use quote::ToTokens;
+use syn::{
+    ext::IdentExt,
+    parse::{Parse, ParseBuffer, Result as ParseResult},
+    spanned::Spanned,
+    token, Ident, LitStr,
+};
+
+syn::custom_punctuation!(DoubleSub, --);
+
+#[derive(Debug, Clone)]
+pub enum IdentPart {
+    Dash(Punct),
+    Ident(Ident),
+}
+
+#[derive(Debug, Clone)]
+pub struct CssIdent {
+    parts: Vec<IdentPart>,
+}
+
+impl IdentPart {
+    pub fn peek(lookahead: &ParseBuffer, accept_dash: bool, accept_ident: bool) -> bool {
+        let peek_dash = accept_dash && (lookahead.peek(token::Sub) || lookahead.peek(DoubleSub));
+        let peek_ident = accept_ident && lookahead.peek(Ident::peek_any);
+        peek_dash || peek_ident
+    }
+}
+
+impl CssIdent {
+    pub fn peek(lookahead: &ParseBuffer) -> bool {
+        if lookahead.peek(token::Sub) {
+            // A single dash is not an identifier
+            lookahead.peek2(token::Sub) || lookahead.peek2(Ident::peek_any)
+        } else {
+            IdentPart::peek(lookahead, true, true)
+        }
+    }
+
+    pub fn stringify(&self) -> String {
+        self.parts
+            .iter()
+            .map(|p| match p {
+                IdentPart::Dash(_) => "-".into(),
+                IdentPart::Ident(t) => format!("{}", t.unraw()),
+            })
+            .collect()
+    }
+
+    pub fn quote_literal(&self) -> LitStr {
+        let formatted = self.stringify();
+        LitStr::new(&formatted, self.span())
+    }
+
+    pub fn quote_at_rule(&self) -> LitStr {
+        let formatted = self.stringify().from_case(Case::Camel).to_case(Case::Kebab);
```

Comment:
Sensible, I was thinking that when one just wants to copy paste over existing code from emotion, it'd be nice to have. You think an error makes sense when non-kebab style is detected?

Reply:
> copy paste over existing code from emotion

You cannot really do that.

Because the Dict / HashMap style of emotion uses comma other than colon and all values are quoted strings.